### PR TITLE
make idle delay configurable

### PR DESCRIPTION
**Pull request by madflow**
* #4 
* In order to pass the `--idle-delay` value down to the worker, I introduced a `WorkerOptions` object. I suspect there could be more options to come.
* I tried to write tests - but no avail so far. I tried to trap the `idleDelay` option by using Jest timer mocks (https://jestjs.io/docs/en/timer-mocks.html) but I could not get it to work as expected.
* Maybe we can agree on the implementation first (or reject it) and then I can tackle the tests again.

### CODE REVIEW

**Comment by madflow** [comment]
Rebased.

**Comment by benjie** [comment]
👍 I’ll try and look at this this week

File: README.md [hunk 1]
```diff
@@ -110,6 +110,24 @@ npx graphile-worker -c "postgres://localhost/mydb"
...
+  --once, -1        Run until there are no runnable jobs left, then exit
+                                                      [boolean] [default: false]
+  --watch, -w       [EXPERIMENTAL] Watch task files for changes, automatically
+                    reloading the task code without restarting worker
+                                                      [boolean] [default: false]
+  --jobs, -j        number of jobs to run concurrently              [default: 1]
+  --idle-delay      how long to wait between polling for jobs in milliseconds
+                                                        [number] [default: 2000]
+```
+
```
Comment (benjie) [inline README.md]: Changed this to use the `graphile-worker --help` output (should make it easier to maintain)

File: __tests__/main.start.test.ts [hunk 2]
```diff
@@ -32,7 +32,7 @@ test("main will execute jobs as they come up, and exits cleanly", () =>
     };
 
     // Run the worker
-    const workerPool = start(tasks, pgPool, 3);
+    const workerPool = start(tasks, pgPool, { workerCount: 3 });
```
Comment (benjie) [inline __tests__/main.start.test.ts]: Undid the changes to most of the tests (except this one) to minimise the diff.

File: src/cli.ts [hunk 3]
```diff
@@ -39,7 +45,17 @@ const isInteger = (n: number): boolean => {
...
+const IDLE = isInteger(argv["idle-delay"]) ? argv["idle-delay"] : IDLE_DELAY;
+
+const workerOptions: WorkerOptions = {
+  idleDelay: IDLE,
+};
+
+const workerPoolOptions: WorkerPoolOptions = {
+  workerCount: JOBS,
+  ...workerOptions
+}
```
Comment (benjie) [inline src/cli.ts]: Differentiated between WorkerPool options and Worker options.

File: src/interfaces.ts [hunk 4]
```diff
@@ -1,6 +1,19 @@
...
+ *
+ * - job: an entry in the `jobs` table, representing work to be done
+ * - queue: an entry in the `job_queues` table, representing a list of jobs to be executed sequentially
+ * - task_identifier: the name of the task to be executed to complete the job
+ * - task: a function representing code to be executed for a particular job (identified by the task_identifier); i.e. a file in the `tasks/` folder
+ * - task list: an object collection of named tasks
+ * - watched task list: an abstraction for a task list that can be updated when the tasks on the disk change
+ * - worker: the thing that checks out a job from the database, executes the relevant task, and then returns the job to the database with either success or failure
+ * - worker pool: a collection of workers to enable processing multiple jobs in parallel
+ */
```
Comment (benjie) [inline src/interfaces.ts]: Added this handy block of terminology which probably ought to be tidied up and put in the README.

**Pull request merged**